packages/studio/src/components/NumberPropInput.tsx

```diff
@@ -0,0 +1,74 @@
+import { PropValueType, TypeGuards } from "@yext/studio-plugin";
+import { ChangeEvent, useCallback, useMemo } from "react";
+import FieldPickerInput from "./FieldPicker/FieldPickerInput";
+import { get } from "lodash";
+import useStudioStore from "../store/useStudioStore";
+import useRawSiteSettings from "../hooks/useRawSiteSettings";
+
+interface PropInputProps {
+  value: number;
+  onChange: (value: number) => void;
+  disabled?: boolean;
+}
+
+/**
+ * Renders the input element of a number prop, that will update the
+ * corresponding prop's value for the active component based on a user's input.
+ */
+export default function NumberPropInput({
+  value,
+  onChange,
+  disabled,
+}: PropInputProps): JSX.Element {
+  const expressionSources = useExpressionSources();
+  const handleChangeEvent = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const val = Number(e.target.value);
+      onChange(val);
+    },
+    [onChange]
+  );
+
+  const appendField = useCallback(
+    (fieldId: string) => {
+      const newPropValue = get(
+        { document: expressionSources["document"] },
```

Comment:
Right now if you use the field picker for say businessId, it'll add the number literal to the prop input like `12345`
However this isn't what we want since when this gets saved the component ends up looking like
```tsx
<MyComponent num={12345}/>
```
instead of
```tsx
<MyComponent num={document.businessId}/>
```

In studio we have a concept of Expressions and Literals via PropValueKind.Expression and PropValueKind.Literal. We need NumberPropInput to be able to handle both kinds. For example maybe the initial value for a number prop is a literal, but when they use the field picker it instead changes it into an expression

Reply:
new [loom](https://www.loom.com/share/038cfc9931fe43dfa40d4b9fcb1d1256?sid=09348a93-b461-457d-a3ca-f25ba186218f). 

Hmmm so I have it working for expressions, but I kind of manually set it as an expression? Not my favorite way to fix it unfortunately, but I couldn't seem to find an earlier / easier way to change the PropValueKind of that component. It does however work + doesn't accept arbitrary strings (only allows strings when set via field dropdown)